src/packages/solid/components/Icon/Icon.tsx

```diff
@@ -49,9 +49,13 @@ const Icon: Component<IconProps> = props => {
   return (
     <View
       {...props}
-      style={props?.style?.Container ?? styles.Container}
       tone={props.tone ?? styles.tone}
-      states={props.tone ?? styles.tone}
+      style={props?.style?.Container ?? styles.Container}
+      {...{
+        ...styles.Container[props.tone || styles.tone],
+        ...props?.style?.Container[props.tone || styles.tone]
+      }}
```

Comment:
@anthony9187 ^ - I'm just saying this is messy. With the new style change you can pass these into the array of styles...

Reply:
totally agree - i hated this setup, and think this super-destructor should be buried and never spoken of again